### PR TITLE
Cody Web: Fix cody web file/symbol mention search

### PR DESCRIPTION
**Pull request by vovakulikov**
Fixes CODY-3375
Fixes CODY-3376

This PR makes symbol and file mention search respect the initial context repository, so we will look for files and symbols only within this repository. You can still select files from different repositories by selecting the "Files" category directly. 

## Test plan
- Check that you can select files in the cody web demo by typing `@<search term>` string and you see files only within sourcegraph/cody repository (which is set as a default context for the cody web demo)

### CODE REVIEW

File: web/lib/components/CodyWebPanelProvider.tsx [hunk 1]
```diff
@@ -114,6 +114,24 @@ export const CodyWebPanelProvider: FunctionComponent<PropsWithChildren<CodyWebPa
         })()
     }, [accessToken, serverEndpoint, createAgentWorker, customHeaders, telemetryClientName])
 
+    useEffect(() => {
```
Comment (vovakulikov) [inline web/lib/components/CodyWebPanelProvider.tsx:117]: Move this to the useEffect to call this `'context/choose-remote-search-repo'` last, cause during chat initialization there are fired some events that reset remote search repo (new session id and restore chat event)

File: web/lib/components/CodyWebPanelProvider.tsx [hunk 2]
```diff
@@ -114,6 +114,24 @@ export const CodyWebPanelProvider: FunctionComponent<PropsWithChildren<CodyWebPa
...
+        }
+
+        // Set initial context after we restore history so context won't be
+        // overridden by the previous chat session context
+        if (initialContext?.repository) {
+            void client.rpc.sendRequest('webview/receiveMessage', {
+                id: activeWebviewPanelID,
+                message: {
+                    command: 'context/choose-remote-search-repo',
+                    explicitRepos: [initialContext?.repository],
```
Comment (thenamankumar) [inline web/lib/components/CodyWebPanelProvider.tsx:129]: `initialContext?.repository ? [initialContext.repository] : []`

File: vscode/src/editor/utils/editor-context.ts [hunk 3]
```diff
@@ -192,7 +192,7 @@ export async function getSymbolContextFiles(
             item.symbols.map(symbol => ({
                 type: 'symbol',
                 remoteRepositoryName: item.repository.name,
-                uri: URI.file(item.repository.name + symbol.location.resource.path),
+                uri: URI.file(`${item.repository.name}/${symbol.location.resource.path}`),
```
Comment (thenamankumar) [inline vscode/src/editor/utils/editor-context.ts:195]: Does `URI.file` correctly handles `//` if they occur in the path?

**Review by thenamankumar — APPROVED** [review]

File: web/lib/components/CodyWebPanelProvider.tsx (hunk 2 above)
Comment (vovakulikov) [reply web/lib/components/CodyWebPanelProvider.tsx]: Just reilized that there is no need for ? operator here since above is if guard statement

**Comment by kalanchan** [comment]
yeahhhhhhhh vova!!!

**Pull request merged**